user_gen.py

```diff
@@ -0,0 +1,139 @@
+import numpy as np
+
+
+def poisson_process(rate=1, size=None):
+    exp_draws = np.random.exponential(scale=rate, size=size)
+    sum_exp_draws = [sum(exp_draws[:i]) for i, x in enumerate(exp_draws)]
+    return sum_exp_draws
+
+
+def poisson_plus(lam=1, plus=1, size=None):
+    return np.random.poisson(lam=lam, size=size) + plus
+
+
+def int_exponential(rate=1, size=None):
+    exp_draws = np.random.exponential(scale=rate, size=size)
+    return np.round(exp_draws)
+
+
+def is_pos_inc_seq(seq):
+    return (sorted(seq) == list(seq) and all([x >= 0 for x in seq]))
```

Comment:
If sequence is sorted you only need to check the first element. There are also a number of improvement to this algorithme if you want to play with this no need to.